### PR TITLE
Fixed NewSimulatedBackend

### PR DESCRIPTION
**Pull request by terenc3t**
With latest Geth, `NewSimulatedBackend` takes in an extra `gasLimit` parameter. This PR fixed that and deleted old SMC tests.

### CODE REVIEW

**Comment by rawfalafel** [comment]
do you know what's causing the error in travis?

**Comment by terenc3t** [comment]
We need to update bazel-go-ethereum with latest geth

File: WORKSPACE [hunk 1]
```diff
@@ -95,8 +95,8 @@ go_repository(
     # code.
     remote = "https://github.com/prysmaticlabs/bazel-go-ethereum",
     vcs = "git",
-    # Last updated July 15, 2018
-    commit = "fad71da72e539ff79183a5d548b105c73ce4969f",
+    # Last updated Augest 15, 2018
```
Comment (prestonvanloon) [inline WORKSPACE]: August

File: beacon-chain/blockchain/BUILD.bazel [hunk 2]
```diff
@@ -29,6 +29,7 @@ go_test(
         "service_test.go",
     ],
     embed = [":go_default_library"],
+    race = "off",  # TODO: fix issues with tests failing with race on.
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/BUILD.bazel]: Please reference the github ID

```
# TODO(12345): ...
```

**Pull request merged**